### PR TITLE
e2e test: test azure disk volume

### PR DESCRIPTION
**Pull request by rootfs**
**What this PR does / why we need it**:
E2E test Azure disk volume
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
@kubernetes/sig-testing-pr-reviews 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by jeffvance** [comment]
@k8s-bot bazel test this
@k8s-bot unit test this

**Comment by rootfs** [comment]
@deads2k can you approve? thanks

**Comment by deads2k** [comment]
> @deads2k can you approve? thanks

Sure, but I'll leave the review to someone else.

/approve

File: hack/ginkgo-e2e.sh [hunk 1]
```diff
@@ -29,6 +29,7 @@ e2e_test=$(kube::util::find-binary "e2e.test")
 # --- Setup some env vars.
 
 GINKGO_PARALLEL=${GINKGO_PARALLEL:-n} # set to 'y' to run tests in parallel
+CLOUD_CONFIG=${CLOUD_CONFIG:-""}
```
Comment (jeffvance) [inline hack/ginkgo-e2e.sh]: is CLOUD_CONFIG used by any other providers?

Comment (rootfs) [inline hack/ginkgo-e2e.sh]: azure is the first but openstack could also use it in the future.

File: test/e2e/framework/test_context.go [hunk 2]
```diff
@@ -209,6 +210,7 @@ func RegisterClusterFlags() {
 	flag.IntVar(&cloudConfig.NumNodes, "num-nodes", -1, "Number of nodes in the cluster")
 
 	flag.StringVar(&cloudConfig.ClusterTag, "cluster-tag", "", "Tag used to identify resources.  Only required if provider is aws.")
+	flag.StringVar(&cloudConfig.ConfigFile, "cloud-config-file", "", "Cloud config file.  Only required if provider is azure.")
```
Comment (jeffvance) [inline test/e2e/framework/test_context.go]: not required for openstack?

File: test/e2e/storage/volumes.go [hunk 3]
```diff
@@ -686,4 +686,54 @@ var _ = framework.KubeDescribe("Volumes [Volume]", func() {
 			framework.TestVolumeClient(cs, config, &fsGroup, tests)
 		})
 	})
+	////////////////////////////////////////////////////////////////////////
+	// Azure Disk
+	////////////////////////////////////////////////////////////////////////
+	framework.KubeDescribe("Azure Disk", func() {
```
Comment (jeffvance) [inline test/e2e/storage/volumes.go]: Do you need "[Feature:Volumes]" in the text? AFAIK, this causes the test to be run in a separate suite and indicates that the test has non-default requirements to run. I think not but just checking.

Comment (rootfs) [inline test/e2e/storage/volumes.go]: good idea!

File: test/e2e/framework/test_context.go (hunk 2 above)
Comment (rootfs) [inline test/e2e/framework/test_context.go]: I haven't plugged it into openstack yet. Once there, I'll update the message.

**Comment by jeffvance** [comment]
@k8s-bot test this

**Comment by rootfs** [comment]
@k8s-bot unit test this

**Comment by rootfs** [comment]
@k8s-bot verify test this

**Pull request merged**